src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -267,8 +269,20 @@ protected internal int GetInvocationCounter()
         /// <returns>The next random number.</returns>
         protected internal BigInteger GetRandom()
         {
-            nonceData = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network);
-            return new BigInteger(nonceData, isUnsigned: true);
+            byte[] buffer;
+            long price;
+            if (IsHardforkEnabled(Hardfork.HF_Aspidochelone))
+            {
+                buffer = Cryptography.Helper.Murmur128(nonceData, random_times++);
```

Comment:
The first entry will be with 0 as a salt, I don't know If it can facilitate any cryptographic attack, but I prefer it to be another value. Bad feelings.

```suggestion
                buffer = Cryptography.Helper.Murmur128(nonceData, ++random_times);
```